d2target/d2target.go

```diff
@@ -72,6 +73,66 @@ type Diagram struct {
 	Steps     []*Diagram `json:"steps,omitempty"`
 }
 
+// boardPath comes in the form of "x/layers/z/scenarios/a"
+// or in the form of "layers/z/scenarios/a"
+func (d *Diagram) GetBoard(boardPath string) *Diagram {
+	path := strings.Split(boardPath, string(os.PathSeparator))
+	if len(path) == 0 || len(boardPath) == 0 {
+		return d
+	}
+
+	head := path[0]
+
+	if head == "index" {
+		return d
+	}
+
+	switch head {
+	case "layers", "scenarios", "steps":
+		if len(path) < 2 {
+			return nil
+		}
+	}
+
+	switch head {
+	case "layers":
+		for _, b := range d.Layers {
+			if b.Name == path[1] {
+				return b.GetBoard(strings.Join(path[2:], string(os.PathSeparator)))
+			}
+		}
+	case "scenarios":
+		for _, b := range d.Scenarios {
+			if b.Name == path[1] {
+				return b.GetBoard(strings.Join(path[2:], string(os.PathSeparator)))
+			}
+		}
+	case "steps":
+		for _, b := range d.Steps {
+			if b.Name == path[1] {
+				return b.GetBoard(strings.Join(path[2:], string(os.PathSeparator)))
+			}
+		}
+	}
+
+	for _, b := range d.Layers {
```

Comment:
Should this be allowed? Why not require the board type prefix?

Reply:
when the root diagram has just `layers`, e.g.

```d2
x

layers: {
  y: { ... }
}
```

then the file dir looks like

```
- y/
  - ...
- index.svg
```

---------

But if the root diagram has multiple board types, like

```d2
x

layers: { ... }
scenarios: { ... }
```

then the directory will look like

```
- layers/
- scenarios/
- index.svg
```

Reply:
Right!